pkg/api/pod/util.go

```diff
@@ -447,6 +449,19 @@ func dropDisabledProcMountField(podSpec, oldPodSpec *api.PodSpec) {
 	}
 }
 
+func dropDisabledFSGroupFields(podSpec, oldPodSpec *api.PodSpec) {
+	if oldPodSpec == nil {
```

Comment:
Shouldn't this be dropped on new objects if the gate is disabled?

Reply:
It does get dropped below if gate is disabled and oldPodSpec did not had the new field. This oldSpec == nil check is there mainly for existing tests that cause panic because they are passing pod with nil spec.

Reply:
When a pod is created, oldPodSpec is nil, so we don’t get to the feature gate check, so the field isn’t dropped?

Or do you have *another* drop method being called that I’m missing?

Reply:
You are right. This return statement should not exist. I have removed it. The `oldPodSpec` check being nil belongs to `fsGroupPolicyInUse` function  not here.

Also, I added additional unit tests that cover this when `oldPodSpec` is nil.